cli/commands/quote-to-ratio.ts

```diff
@@ -0,0 +1,160 @@
+/// <reference types="./types/bunyan-debug-stream" />
+import { flags } from '@oclif/command';
+import { Currency, Ether, Percent } from '@uniswap/sdk-core';
+import { Position } from '@uniswap/v3-sdk';
+import dotenv from 'dotenv';
+import { ethers } from 'ethers';
+import { ID_TO_CHAIN_ID, parseAmount, SwapRoute } from '../../src';
+import { BaseCommand } from '../base-command';
+
+dotenv.config();
+
+ethers.utils.Logger.globalLogger();
+ethers.utils.Logger.setLogLevel(ethers.utils.Logger.levels.DEBUG);
+
+export class QuoteToRatio extends BaseCommand {
+  static description = 'Uniswap Smart Order Router CLI';
+
+  static flags = {
+    ...BaseCommand.flags,
+    version: flags.version({ char: 'v' }),
+    help: flags.help({ char: 'h' }),
+    token0: flags.string({ char: 'i', required: true }),
+    token1: flags.string({ char: 'o', required: true }),
+    feeAmount: flags.integer({ char: 'f', required: true }),
+    token0Balance: flags.string({ required: true }),
+    token1Balance: flags.string({ required: true }),
+    recipient: flags.string({ required: true }),
+    tickLower: flags.integer({ required: true }),
+    tickUpper: flags.integer({ required: true }),
+  };
+
+  async run() {
+    const { flags } = this.parse(QuoteToRatio);
+    const {
+      chainId: chainIdNumb,
+      token0: token0Str,
+      token1: token1Str,
+      token0Balance: token0BalanceStr,
+      token1Balance: token1BalanceStr,
+      feeAmount,
+      tickLower,
+      tickUpper,
+      recipient,
+      debug,
+      topN,
+      topNTokenInOut,
+      topNSecondHop,
+      topNWithEachBaseToken,
+      topNWithBaseToken,
+      topNWithBaseTokenInSet,
+      maxSwapsPerPath,
+      maxSplits,
+      distributionPercent,
+    } = flags;
+
+    const log = this.logger;
+    const router = this.swapToRatioRouter;
+    const tokenProvider = this.tokenProvider;
+
+    const tokenAccessor = await tokenProvider.getTokens([
+      token0Str,
+      token1Str,
+    ]);
+
+    const chainId = ID_TO_CHAIN_ID(chainIdNumb);
+    const tokenIn: Currency =
+      token0Str == 'ETH'
+        ? Ether.onChain(chainId)
+        : tokenAccessor.getTokenByAddress(token0Str)!;
+    const tokenOut: Currency =
+      token1Str == 'ETH'
+        ? Ether.onChain(chainId)
+        : tokenAccessor.getTokenByAddress(token1Str)!;
+
+    const tokenInBalance = parseAmount(token0BalanceStr, tokenIn);
+    const tokenOutBalance = parseAmount(token1BalanceStr, tokenOut);
+
+    const poolAccessor = await this.poolProvider?.getPools([
```

Comment:
Shouldn't need the `?.` anymore now that you throw errors in the getters, here and below